lib/components/Input/CheckboxInput.module.scss

```diff
@@ -1,65 +1,80 @@
 @import '../../common/constants';
 @import '../../common/mixins';
 
-$line-height: 4*$grid-size;
 $line-height-small: 3*$grid-size;
-$outline-focus-border-width: 1px;
 
-.checkbox-container {
-    color: var(--color-text-rest);
+.checkbox-button {
+    @include md-box(inline-block, relative);
+    width: 4*$grid-size;
+    height: 4*$grid-size;
+    background-color: var(--color-bg-input-rest);
+    outline: none;
+    vertical-align: top;
+    border-radius: $component-border-radius;
+    transition: background-color 0.2s ease-out;
+    border: 1px solid var(--color-border-rest);
+}
 
-    &:not(.disabled) input:focus + .checkbox-button {
-        outline: $outline-focus-border-width dashed var(--color-border-focus);
-        outline-offset: -$outline-focus-border-width;
-    }
+.checkbox-text {
+    @include md-box(inline-block, relative);
+    vertical-align: top;
+    padding-left: $gutter-xxsmall;
 
-    .checkbox-button {
-        border: 1px solid var(--color-border-rest);
-        background-color: var(--color-bg-input-rest);
+    @include rtl {
+        padding-left: unset;
+        padding-right: $gutter-xxsmall;
     }
+}
 
-    &.disabled {
-        color: var(--color-text-disabled);
-        .checkbox-button {
-            border: 1px solid var(--color-border-disabled);
-            background-color: var(--color-bg-input-rest);
-        }
-    }
-    .checkbox-fill {
-        background-color: var(--color-text-rest);
+.checkbox-fill {
+    @include md-box(display-none, absolute);
+    left: 1.5*$grid-size;
+    top: 1.5*$grid-size;
+    width: $grid-size;
+    height: $grid-size;
+    background-color: var(--color-text-rest);
+}
+
+.checkbox-label {
+    @include md-box(inline-block, relative);
+}
+
+.checkbox-checkmark {
+    @include md-box(display-none, absolute);
+    line-height: var(--line-height-xsmall);
+    color: var(--color-border-hover);
+    left: 0;
+    cursor: pointer;
+
+    @include rtl {
+        left: unset;
+        right: 0;
     }
 }
 
 .checkbox-container:hover {
-    .checkbox-button {
-        border: 1px solid var(--color-border-checkbox-hover) !important;
+    &:not(.disabled):not(.selected):not(.indeterminate) .checkbox-button {

```

Comment:
nit: for a bit more readability, please put spaces between each css pseudo-class here and everywhere else on this file (but only if css syntax allows for it. if css requires us to do this with no spaces, then resolve this comment)

Reply:
in css spaces will mean different things.... if you do `&:not(.disabled):not(.selected)` it means the current element when it doesn't have the selected or disabled class.... if you do `& :not(.disabled) :not(.selected)` it means every element that doesn't have selected class inside an element without the disabled class inside the current element.

Reply:
good to know thanks for the clarification